crates/pop-cli/src/commands/up/contract.rs

```diff
@@ -124,16 +125,18 @@ impl UpContractCommand {
 				},
 			};
 		}
-		let spinner = cliclack::spinner();
-		spinner.start("Uploading and instantiating the contract...");
-		let contract_address = instantiate_smart_contract(instantiate_exec, weight_limit)
-			.await
-			.map_err(|err| anyhow!("{} {}", "ERROR:", format!("{err:?}")))?;
-		spinner.stop(format!(
-			"Contract deployed and instantiated: The Contract Address is {:?}",
-			contract_address
-		));
-		outro("Deployment complete")?;
+		if !self.dry_run {
```

Comment:
What if it is a dry_run?

Reply:
If it's a dry-run, it displays the required weight without executing any blockchain operations. Everything within this if clause is not executed.